src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -522,23 +592,68 @@ class InstructionRenderer extends Disposable implements ITableRenderer<IDisassem
 	}
 
 	renderTemplate(container: HTMLElement): IInstructionColumnTemplateData {
+		const sourcecode = append(container, $('.sourcecode'));
 		const instruction = append(container, $('.instruction'));
+		this.applyFontInfo(sourcecode);
 		this.applyFontInfo(instruction);
-
 		const currentElement: { element?: IDisassembledInstructionEntry } = { element: undefined };
 
 		const disposables = [
-			this._disassemblyView.onDidChangeStackFrame(() => this.rerenderBackground(instruction, currentElement.element))
+			this._disassemblyView.onDidChangeStackFrame(() => this.rerenderBackground(instruction, sourcecode, currentElement.element)),
+			addStandardDisposableListener(sourcecode, 'click', () => this.openSourceCode(currentElement.element?.instruction!)),
 		];
 
-		return { currentElement, instruction, disposables };
+		return { currentElement, instruction, sourcecode, disposables };
 	}
 
 	renderElement(element: IDisassembledInstructionEntry, index: number, templateData: IInstructionColumnTemplateData, height: number | undefined): void {
-		templateData.currentElement.element = element;
+		this.renderElementInner(element, index, templateData, height);
+	}
 
+	private async renderElementInner(element: IDisassembledInstructionEntry, index: number, templateData: IInstructionColumnTemplateData, height: number | undefined): Promise<void> {
+		templateData.currentElement.element = element;
 		const instruction = element.instruction;
-		const sb = createStringBuilder(10000);
+		templateData.sourcecode.innerText = '';
+		const sb = createStringBuilder(1000);
+
+		if (this._disassemblyView.isSourceCodeRender && instruction.location?.path && instruction.line) {
+			let sourceURI = this.getUriFromSource(instruction);
+
+			if (sourceURI) {
+				let textModel: ITextModel | undefined = undefined;
+				const sourceSB = createStringBuilder(10000);
+
+				try {
+					const ref = await this._disassemblyView.textModelService.createModelReference(sourceURI);
+					textModel = ref.object.textEditorModel;
+					templateData.disposables.push(ref);
+
+				} catch {
+					const textFileContent = await this._disassemblyView.nativeTextFileService.read(sourceURI);
```

Comment:
Not sure when it would fail, I copied this section from "bulkEditTree.ts".  Both codepath works.  My guess is that textModelService does some caching and keepalive.

Reply:
You've skipped the part about disposing the textmodel which is an important part :) @jrieken is there any reason to have this try/catch on `createModelReference` and manually instantiate a TextModel?

Reply:
Ping.  @jrieken